source/TemplateProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.IO;
+using System.Text;
+
+namespace VersionOne.Localization
+{
+	/*
+	 * Replacement for the ITemplateSetLoader to allow for an enumeration of individual source endpoints
+	 * that can be mixed in any order between files and database tables.
+	 */
+	public interface ITemplateProvider
+	{
+		ITemplateSet Load(string culture);
+
+		string Name { get; }
+	}
+
+	public class CompatibilityTemplateProvider : ITemplateProvider
+	{
+		private readonly ITemplateSetLoader _loader;
+
+		public CompatibilityTemplateProvider(ITemplateSetLoader loader, string setname)
+		{
+			_loader = loader;
+			Name = setname;
+		}
+
+		public ITemplateSet Load(string culture)
+		{
+			return _loader.Load(culture, Name);
+		}
+
+		public string Name { get; private set; }
+	}
+
+	public class FileTemplateProvider : ITemplateProvider
```

Comment:
`FileTemplateProvider` is superfluous, since it just duplicates the code from existing `FileTemplateSetLoader`, and its functionality is simply a binding of `FileTemplateSetLoader` to a set name via `CompatibilityTemplateProvider` instance.